### PR TITLE
check-gateway: Improved gateway check

### PR DESCRIPTION
**Pull request by sargon**
The improvements include checks for tun interface, openvpn and multiple
ping targets allowed. It is recommended that GW_CONTROL_IP is extended
with several targets. Also the new version will put the system
into maintenance iff it can't repair the situation within (configurable) 10
minutes.

### CODE REVIEW

File: files/usr/local/bin/check-gateway [hunk 1]
```diff
@@ -2,19 +2,26 @@
...
+
 . /etc/ffnord
 
 MAINTENANCE=${MAINTENANCE:-0}
+GW_INTERFACE=${GW_INTERFACE:-"tun-anonvpn"}
+MAX_INTERFACE_MISSING=${MAX_INTERFACE_MISSING:-10}
+MAX_PKT_LOSS=${MAX_PKT_LOSS:-50}
+MAX_PKT_LOSS_EXCEED=${MAX_PKT_LOSS_EXCEED:-2}
+INTERFACE_MISSING_STATE="/tmp/chkgw-interface-missing"
+PKT_LOSS_EXCEEDED_STATE="/tmp/chkgw-pktloss-exceeded"
```
Comment (rubo77) [inline files/usr/local/bin/check-gateway]: please add a small comment above each config value

File: manifests/resources/checkgw.pp [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class ffnord::resources::checkgw (
-  $gw_control_ip     = "8.8.8.8",     # Control ip addr
+  $gw_control_ip     = "8.8.8.8",     # Control ip addr, you want to change this!
```
Comment (rubo77) [inline manifests/resources/checkgw.pp]: why? lets add a better default IP here, that works right away

**Comment by sargon** [comment]
Hmpf. Okay, then we have to remove that line and enforce input from the manifest.
I choose another set on the kiel gateways anyway.

**Comment by sargon** [comment]
So this should do the job, please test before merging. I had puppet to checkrun on my dev machine.

**Comment by sargon** [comment]
Ich merge das dann mal.

**Pull request merged**